### DISCUSSION TITLE
did aiogram have their own handling exception?

### DISCUSSION THREAD

**Discussion by sinkaroid** [aiogram/aiogram#540] [Q&A] [answered]
did aiogram have their own handling exception?
Example like : [https://github.com/python-telegram-bot/python-telegram-bot/wiki/Exception-Handling](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Exception-Handling)  
or we need to catch manualy?  

sorry if i mentioned other libs 
I will close this discuss immediately when I get any answer, thankyou sir

**Comment by evgfilim1** [comment, marked as answer]
Hello, @sinkaroid!

Aiogram has similar exception handler as in _python-telegram-bot_. You should use `@dp.errors_handler()` or `@dp.errors_handler(exception=ExceptionBaseClass)` to catch all exceptions or specified one respectively. Example:
https://github.com/aiogram/aiogram/blob/15a6e94135f86777af1ff96a7e40c1e5923558e3/examples/callback_data_factory_simple.py#L63-L65

The base exception is [`TelegramAPIError`](https://github.com/aiogram/aiogram/blob/15a6e94135f86777af1ff96a7e40c1e5923558e3/aiogram/utils/exceptions.py#L106), so if you want to handle all Telegram exceptions, use this class. Feel free to ask for more details if needed.

P.S. It's ok if you just mention other libs, no need to worry in your case.

**Comment by sinkaroid** [comment]
Many thanks!
